src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptionConverters.java

```diff
@@ -234,6 +234,42 @@ public String getTypeDescription() {
     }
   }
 
+  /**
+   * Flag converter for canonicalizing a label (possibly with a "/..." suffix) and/or define by
+   * converting the label to ambiguous canonical form.
+   */
+  public static class CustomFlagConverter implements Converter<String> {
+    @Override
+    public String convert(String input, Object conversionContext) throws OptionsParsingException {
+      if (!input.startsWith("//") && !input.startsWith("@")) {
+        // This is a --define flag.
+        return input;
+      }
+      // A "/..." suffix is not valid label syntax, so replace it with valid syntax and transform
+      // it back after conversion.
+      String invalidSubpackagesSuffix = "/...";
+      String validSubpackagesSuffix = ":__subpackages__";
+      String escapedUnconvertedLabel =
+          input.endsWith(invalidSubpackagesSuffix)
+              ? input.substring(0, input.length() - invalidSubpackagesSuffix.length())
```

Comment:
Instead of the `substring` call, use `String.replace`. Any instance of `/...` inside the label is invalid anyway.

Reply:
Wouldn't `//...super_hidden_folder:foo` be a valid label? I can still do this, but I'm not sure it's theoretically correct.

Reply:
Fair enough, I hope no-one is naming packages like that but it's technically possible, so leave this as-is.